### DISCUSSION TITLE
Realtime filtering with events API

### DISCUSSION THREAD

**Discussion by saihaj** [farcasterxyz/protocol#168] [FIP Stage 1: Ideas]
Events API allows me to access realtime event data. However, filtering this real time data remains a responsibility of the application layer. Enhancements to the Events API would be beneficial, such as:
- Introducing a filter by hash/parent hash feature akin to `eth_getFilterLogs`, enabling targeted monitoring of actions within a cast.
- Filter by FIDs, facilitating the tracking of events for specific groups of users.
- Filter by parentUrl, enabling the monitoring of events occurring on a particular channel

**Comment by vrypan** [comment]
How is this different compared to the gRPC `Subscribe(SubscribeRequest)` method? Can you elaborate?

**Reply by saihaj** [reply]
These new filtering options would extend the existing `Subscribe(SubscribeRequest)` gRPC method. Today SubscribeRequest is limited to types of events to subscribe to and a starting event ID.
